components/raftstore/src/store/fsm/peer.rs

```diff
@@ -688,53 +689,52 @@ where
                 }
             }
         }
-        // Propose batch request which may be still waiting for more raft-command
-        if self.ctx.sync_write_worker.is_some() {
-            self.propose_batch_raft_command(true);
-        } else {
-            self.propose_batch_raft_command(false);
-            self.check_batch_cmd_and_proposed_cb();
-        }
+        self.on_loop_finished();
     }
 
-    fn propose_batch_raft_command(&mut self, force: bool) {
-        if self.fsm.batch_req_builder.request.is_none() {
-            return;
-        }
-        if !force
-            && self.ctx.cfg.cmd_batch_concurrent_ready_max_count != 0
-            && self.fsm.peer.unpersisted_ready_len()
-                >= self.ctx.cfg.cmd_batch_concurrent_ready_max_count
-        {
-            return;
+    #[inline]
+    #[allow(unused_mut, clippy::redundant_closure_call)]
+    fn on_loop_finished(&mut self) {
+        let mut ready_concurrency = self.ctx.cfg.cmd_batch_concurrent_ready_max_count;
+        let mut should_propose = self.ctx.sync_write_worker.is_some()
+            || ready_concurrency == 0
+            || self.fsm.peer.unpersisted_ready_len() < ready_concurrency;
+        #[cfg(feature = "failpoints")]
```

Comment:
why? `fail_point!` would check it innerly.

Reply:
It's just a habit of mine. Even though the empty function call will be optimized away, but I feel uncomfortable to leave a trail in production code.

Reply:
it's not neat and of no use.

Reply:
I don't think it's no use though. My brain can filter out `#[cfg(feature = "failpoints")]` and `fail_point!(..)` for me. But it has to pause a second for the nested closure call.

BUT, considering it has become a common pattern in TiKV's codebase, I'll keep it consistent.